### PR TITLE
Do not suppress panic exception in runtime type check

### LINKED ISSUE #7761: Typo in an argument results in a Type_Error
### Discord username

_No response_

### What type of issue is this?

Permanent – Occurring repeatably

### Is this issue blocking you from using Enso?

- [ ] Yes, I can't use Enso because of this issue.

### Is this a regression?

- [ ] Yes, previous version of Enso did not have this issue.

### What issue are you facing?

A typo in `operator13.rename_columns (Maps.from_Vector [["Sum Right Matches Played in Tournament", "previous_tournament_matches"]])` (I used `Maps.from_Vector` instead of `Map.from_vector`) produced only a `Type_Error`.

### Expected behaviour

I expect it to say `Maps not found` or something like that.

### How we can reproduce it?

1. Run local IDE against cloud backend
2. Create a node with the text shown above
3. See `Type_Error`

### Screenshots or screencasts

_No response_

### Logs

_No response_

### Enso Version

local build on branch wip/gmt/5105-post

### Browser or standalone distribution

Standalone distribution (cloud project)

### Browser Version or standalone distribution

standalone

### Operating System

MacOS

### Operating System Version

Monterey 12.6.3

### Hardware you are using

MacBook Air (M1, 2020)

### PR DESCRIPTION
**Pull request by Akirathan**
Fixes #7761

### Pull Request Description

Instead of `Type_Error`, which does not make sense at that position, display the underlying "Compiler error", like so:

image

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [X] The documentation has been updated, if necessary.
- [X] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [X] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [X] Unit tests have been written where possible.
  - [X] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: engine/runtime/src/main/java/org/enso/interpreter/node/callable/argument/ReadArgumentCheckNode.java [hunk 1]
```diff
@@ -73,6 +74,11 @@ public static boolean isWrappedThunk(Function fn) {
     return false;
   }
 
+  @Specialization
+  Object doPanicSentinel(VirtualFrame frame, PanicSentinel panicSentinel) {
```
Comment (Akirathan) [inline engine/runtime/src/main/java/org/enso/interpreter/node/callable/argument/ReadArgumentCheckNode.java:221]: This follows the same semantics as in [InvokeMethodNode.doPanicSentinel](https://github.com/enso-org/enso/blob/3b3dbc352d1a4726143ff3b78e186edf7cef3143/engine/runtime/src/main/java/org/enso/interpreter/node/callable/InvokeMethodNode.java#L292)

**Review by JaroslavTulach — APPROVED** [review]
Ááá, `PanicSentinel` thank you for the fix. If I may have a wish, create another test in `SignatureTest` - I run `runtime-with-instruments` tests only when I really must as they are too asynchronous.

**Comment by JaroslavTulach** [comment]
But yeah, there is just a single test for [PanicSentinel in runtime](https://github.com/enso-org/enso/blob/ea5096a2880e4819234840487c5820be1228ad16/engine/runtime/src/test/java/org/enso/interpreter/node/callable/SequenceLiteralNodeTest.java#L16).

**Comment by Akirathan** [comment]
> Ááá, `PanicSentinel` thank you for the fix. If I may have a wish, create another test in `SignatureTest` - I run `runtime-with-instruments` tests only when I really must as they are too asynchronous.

A new test added to `SignatureTest` in d772c24704450363b62dff94c6e93c9af0c33d60. I also did few refactorings in SignatureTest - [Do not use hardcoded string](f9fdabd6f9b705699a41eafb0a8c186b1187bf7a) and [Redirect out and err to dev null](6a379f63214a3ff3003a4b1838e59787804b84c5).

File: engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/EnsoProjectNode.java [hunk 2]
```diff
@@ -100,7 +101,7 @@ public Object getCurrentProjectDescr(Object nothing) {
       }
       cachedProjectDescr = cachedProjectDescrRef.get();
     }
-    assert cachedProjectDescr != null;
+    Objects.requireNonNull(cachedProjectDescr);
```
Comment (hubertp) [inline engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/EnsoProjectNode.java:101]: `cachedProjectDescrRef` is a weak reference so wouldn't be surprised if the result can be gone at any point. Can't we just store the result of `createProjectDescriptionAtom` in a variable and return it and leave the rest as-is?

File: engine/runtime/src/main/java/org/enso/interpreter/node/callable/argument/ReadArgumentCheckNode.java (hunk 1 above)
Comment (JaroslavTulach) [reply engine/runtime/src/main/java/org/enso/interpreter/node/callable/argument/ReadArgumentCheckNode.java:221]: OK, but: Another option is to include `PanicSentinel` in [isAllFitValue](https://github.com/enso-org/enso/blob/8a9a345266d138b99a80ef6a0a1fd405c6bafede/engine/runtime/src/main/java/org/enso/interpreter/node/callable/argument/ReadArgumentCheckNode.java#L251) - I believe the result would be the same and the logic of `InvokeMethodNode` wouldn't need to be duplicated. If `PanicSentinel` was `isAllFitValue`, then the `ReadArgumentCheckNode` would be just a _no-op_ and the `PanicSentinel` would then be processed later as in not ascribed situations.

File: engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/EnsoProjectNode.java (hunk 2 above)
Comment (Akirathan) [reply engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/EnsoProjectNode.java:101]: Good idea. Done in 6acf1541bd922ed42f2787fa7cca494bb845b310

**Pull request merged**